client/state/selectors/is-requesting-media.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import MediaQueryManager from 'lib/query-manager/media';
+
+/**
+ * Returns true if media is being requested for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {bool}           True if media is being requested
+ */
+export default function isRequestingMedia( state, siteId, query ) {
+	const queryRequests = state.media.queryRequests[ siteId ];
+	const stringifiedQuery = MediaQueryManager.QueryKey.stringify( query );
+
+	if ( ! queryRequests ) {
+		return false;
+	}
+
+	return queryRequests.indexOf( stringifiedQuery ) !== -1;
```

Comment:
>Hm, according to the documentation, it does not accept a boolean collection argument.

Sorry, I should have been more clear. Since the issue is we're trying to avoid calling `indexOf` when `queryRequests` is potentially `null`, I meant to point out that `_.includes` will automatically account for this and return `false`.

FWIW, though not documented, I did test and confirm that it behaves as expected if passed a boolean:

```
n_ > _.includes( false, 'foo' );
false
```